File: php/testcode/weakhash_003.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakhash_sha1_token
function weakhash003(BenchmarkRequest $req): BenchmarkResponse {
    $token = $req->param('token');
    $hashed = sha1($token); // vuln-code-snippet vuln-line php_weakhash_sha1_token
    return BenchmarkResponse::json(['token_hash' => $hashed]);
}
// vuln-code-snippet end php_weakhash_sha1_token

File: ruby/testcode/weakhash_003.rb
require_relative 'shared'
require 'digest'

# vuln-code-snippet start ruby_weakhash_sha1
def generate_auth_token(req)
  password = req.post('password')
  token = Digest::SHA1.hexdigest(password) # vuln-code-snippet vuln-line ruby_weakhash_sha1
  BenchmarkResponse.ok(token)
end
# vuln-code-snippet end ruby_weakhash_sha1
